pkg/volume/plugins.go

```diff
@@ -121,6 +122,9 @@ type VolumeHost interface {
 	// the provided spec.  See comments on NewWrapperBuilder for more
 	// context.
 	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
+
+	//Get cloud provider from kubelet
+	GetCloudProvider() cloudprovider.Interface
```

Comment:
This is not about volume config but cloud-provider config.  Kubelet has a
flag for cloud-config.  Volumes should not be replicating that.

I think the right answer is to as the VolumeHost for the current
CloudProvder.

On Fri, Aug 21, 2015 at 6:06 AM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/volume/plugins.go
> <https://github.com/kubernetes/kubernetes/pull/6689#discussion_r37631709>:
>
> > @@ -121,6 +122,9 @@ type VolumeHost interface {
> >  	// the provided spec.  See comments on NewWrapperBuilder for more
> >  	// context.
> >  	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
> > +
> > +	//Get cloud provider from kubelet
> > +	GetCloudProvider() cloudprovider.Interface
>
> The passing of configs should not be copied all over.
>
> @thockin <https://github.com/thockin> can you explain this further? I
> proposed volumeConfig in #9870
> <https://github.com/kubernetes/kubernetes/pull/9870> which is extensible
> to any type of config a volume plugin would need. I see no reason it
> couldn't pass an OpenStack config in it. Each volume takes what it needs
> from the volumeConfig struct.
>
> Alternatively, volumeConfig as a struct can be nuked and all config
> options are passed via VolumeHost. I originally did it that way but was
> told that passing config down into components was the preferred method.
>
> I think we should adopt 1 way and use it for all volume configuration.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/6689/files#r37631709>.
>